include-refactoring/xsimd/arch/xsimd_arm7.hpp

```diff
@@ -0,0 +1,58 @@
+#ifndef XSIMD_ARM7_HPP
+#define XSIMD_ARM7_HPP
+
+#include <tuple>
+
+#include "../types/xsimd_arm7_register.hpp"
+#include "../types/xsimd_utils.hpp"
+
+namespace xsimd
+{
+    namespace kernel
+    {
+        using namespace types;
+
+        namespace detail
+        {
+            template <class... T>
+            struct arm_dispatcher
+            {
+                using container_type = std::tuple<T (*)(T, T) ...>;
+                const container_type m_func;
+
+                template <class U>
+                U run(U lhs, U rhs) const
+                {
+                    using func_type = U (*)(U, U);
```

Comment:
this definition would be used in the container_type -> put it in a using definition at struct scope